src/qml/MapCanvas.qml

```diff
@@ -259,9 +262,11 @@ Item {
         mapArea.rightClicked(Qt.point(eventPoint.position.x, eventPoint.position.y), "touch");
       } else {
         if (!doublePressed) {
+          mapArea.clicked(point.position, "touch");
           timer.tapPoint = Qt.point(eventPoint.position.x, eventPoint.position.y);
           timer.restart();
         } else {
+          mapArea.doubleClicked(point.position, "touch");
           mapCanvasWrapper.zoom(Qt.point(eventPoint.position.x, eventPoint.position.y), 0.8);
```

Comment:
You will need to move the zoom action to sit outside of the mapcanvas QML, otherwise the zoom will always be triggered even when the doubleClicked action has been taken over.